Mathlib/Combinatorics/SimpleGraph/DegreeSum.lean

```diff
@@ -52,9 +50,6 @@ section DegreeSum
 
 variable [Fintype V] [DecidableRel G.Adj]
 
--- Porting note: Changed to `Fintype (Sym2 V)` to match Combinatorics.SimpleGraph.Basic
-variable [Fintype (Sym2 V)]
```

Comment:
Why this change? A priori you're making the lemma strictly less general.

Reply:
It's redundant – everything still compiles _without_ it. So the following lemmas cannot have been made less general by the removal of this variable.

Reply:
That's not necessarily true, unfortunately; though certainly I would have expected a docstring explaining whether this is intentional.

Reply:
@kmill, what's your opinion here?

Reply:
@Parcly-Taxel The rule of thumb with Fintype and Decidable instances is that you want to be sure that you have every instance be its own argument, even if that means redundancy. That's to help with all the overlapping instances we tend to have with these classes (especially since sometimes it's intentional, for example with `classical` and `Decidable`).

However -- Lean 4 doesn't seem to play as well with overlapping instances as in Lean 3 (it wants to resynthesize every instance from scratch, but Lean 3 was happy to use instances it picked up from unification), so this rule of thumb seems like it needs tweaking. It sounds plausible to me that if one `[...]` argument can be derived from the others, then there's no difference between including it or excluding it, so excluding it makes sense.

If we care about maximal rewritability, then we should be using `{...}` arguments for instances that can be picked up from unification. That's not great, though, because that tends to mean you need two versions of rewrite lemmas, since different instances tend to be picked up in the forwards and backwards directions. I wonder if we can convince core Lean to add "`[{...}]`" arguments that give us the Lean 3 behavior?

In short, ideally we'd have the extra `Fintype (Sym2 V)` argument for smoother rewriting, but at the moment I'm not sure it actually does that. I could be wrong though.

Reply:
Let's not remove it for the moment then. It'll be much harder to find which lemmas could be generalised by having redundant arguments if those lemmas do not already have those arguments.

Reply:
That seems reasonable to me.

To give a concrete issue that having an extra argument is meant to solve: you can obtain a `Fintype (Sym2 V)` instance from both [`Sym2.instFintype`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Sym.html#Sym2.instFintype) and [`Quotient.fintype`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Basic.html#Quotient.fintype)

Reply:
Reverted.